From dbfd5171ce2a6378cfc51ca9c4326a90b861d694 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Sat, 31 Mar 2018 23:05:27 -0700
Subject: [PATCH] drm/msm/dsi: Fix Clang enum-conversion warning in
 dsi_get_cmd_fmt

Clang warns this value doesn't match what the function's return
type is and it's right. This appears to be a copy paste error
from dsi_get_vid_fmt.

../drivers/gpu/drm/msm/dsi/dsi_host.c:743:35: warning: implicit
conversion from enumeration type 'enum dsi_vid_dst_format' to different
enumeration type 'enum dsi_cmd_dst_format' [-Wenum-conversion]
        case MIPI_DSI_FMT_RGB666:       return VID_DST_FORMAT_RGB666;
                                        ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~

Fixes: a689554ba6ed ("drm/msm: Initial add DSI connector support")
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/gpu/drm/msm/dsi/dsi_host.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/msm/dsi/dsi_host.c b/drivers/gpu/drm/msm/dsi/dsi_host.c
index 0f7324a686ca..d729b2b4b66d 100644
--- a/drivers/gpu/drm/msm/dsi/dsi_host.c
+++ b/drivers/gpu/drm/msm/dsi/dsi_host.c
@@ -740,7 +740,7 @@ static inline enum dsi_cmd_dst_format dsi_get_cmd_fmt(
 	switch (mipi_fmt) {
 	case MIPI_DSI_FMT_RGB888:	return CMD_DST_FORMAT_RGB888;
 	case MIPI_DSI_FMT_RGB666_PACKED:
-	case MIPI_DSI_FMT_RGB666:	return VID_DST_FORMAT_RGB666;
+	case MIPI_DSI_FMT_RGB666:	return CMD_DST_FORMAT_RGB666;
 	case MIPI_DSI_FMT_RGB565:	return CMD_DST_FORMAT_RGB565;
 	default:			return CMD_DST_FORMAT_RGB888;
 	}
-- 
2.16.3

